lib/secure_headers.rb

```diff
@@ -45,181 +100,158 @@ def configure &block
   class << self
     def append_features(base)
       base.module_eval do
-        extend ClassMethods
         include InstanceMethods
       end
     end
 
-    def header_hash(options = nil)
+    # Strips out headers not applicable to this request
+    def header_hash_for(request)
+      unless request.scheme == HTTPS
+        secure_headers_request_config(request)[SecureHeaders::StrictTransportSecurity::CONFIG_KEY] = SecureHeaders::OPT_OUT
+        secure_headers_request_config(request)[SecureHeaders::PublicKeyPins::CONFIG_KEY] = SecureHeaders::OPT_OUT
+      end
+
       ALL_HEADER_CLASSES.inject({}) do |memo, klass|
-        config = if options.is_a?(Hash) && options[klass::Constants::CONFIG_KEY]
-          options[klass::Constants::CONFIG_KEY]
+        header_config = request.env[klass::CONFIG_KEY] ||
+          secure_headers_request_config(request)[klass::CONFIG_KEY]
```

Comment:
Are these two separate API's for overriding the config per-request? If so, it would be nice to just pick one to go with for simplicity's sake.